lib/main.js

```diff
@@ -274,23 +265,25 @@ export default {
 
         let match = parseRegex.exec(result);
         while (match !== null) {
-          // Note that these positions are being converted to 0-indexed
-          const line = Number.parseInt(match[1], 10) - 1 || 0;
-          const col = Number.parseInt(match[2], 10) - 1 || undefined;
+          const line = Number.parseInt(match[1], 10);
+          const col = Number.parseInt(match[2], 10);
 
           const isErr = (match[4] === 'E' && !this.pycodestyleErrorsToWarnings)
             || (match[4] === 'F' && this.flakeErrors);
 
           try {
             messages.push({
-              type: isErr ? 'Error' : 'Warning',
-              text: `${match[3]} — ${match[5]}`,
-              filePath,
-              range: helpers.generateRange(textEditor, line, col),
+              severity: isErr ? 'error' : 'warning',
+              excerpt: `${match[3]} — ${match[5]}`,
+              location: {
+                file: filePath,
+                position: helpers.generateRange(textEditor, line - 1, col - 1),
+              },
             });
           } catch (point) {
             // generateRange encountered an invalid point
-            messages.push(generateInvalidPointTrace(execPath, match, filePath, textEditor, point));
+            const m = generateInvalidPointTrace(execPath, match, filePath, textEditor, line, col);
```

Comment:
The same applies here, the existing code is correctly handling errors, while your changes make it report incorrect data (what is _thought_ to be the point here may not end up being the point returned if it wasn't correct in the first place!).

Reply:
Thanks for the reply! My problem here was I could not validate the `catch (point)` part. What it throws is a string according to my tests. Can you confirm it throws a `Point`?

Reply:
Hmmm, it looks like it does `throw new Error`, so I'm not sure how that is supposed to work either. I won't be able to test anything till later tonight myself.